src/main/java/com/project/saw/event/EventService.java

```diff
@@ -0,0 +1,92 @@
+package com.project.saw.event;
+
+
+import com.project.saw.dto.event.CreateEventRequest;
+import com.project.saw.dto.event.UpdateEventRequest;
+import com.project.saw.dto.event.UpdateEventResponse;
+import com.project.saw.exception.DuplicateException;
+import jakarta.persistence.EntityNotFoundException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.Comparator;
+import java.util.List;
+
+import java.util.Optional;
+
+
+@Slf4j
+@Service
+public class EventService {
+
+    private final EventRepository eventRepository;
+    private static final String DUPLICATE_EVENT_ERROR_MESSAGE = "This event %s already exist";
+
+    public EventService(EventRepository eventRepository) {
+        this.eventRepository = eventRepository;
+    }
+
+    public List<EventEntity> getEventList() {
+        return eventRepository.findAll().stream()
+                .sorted(Comparator.comparing(EventEntity::getStartingDate))
+                .filter(e -> e.getEndingDate().isAfter(LocalDate.now()))
+                .toList();
+    }
+
+    public EventEntity createEvent(CreateEventRequest request) {
+        eventRepository.findByTitleIgnoreCase(request.getTitle())
+                .ifPresent(EventEntity -> {
+                    var error = String.format(DUPLICATE_EVENT_ERROR_MESSAGE, request.getTitle());
+                    throw new DuplicateException(error);
+                });
+        EventEntity eventEntity = EventEntity.builder()
+                .title(request.getTitle())
+                .location(request.getLocation())
+                .price(request.getPrice())
+                .startingDate(request.getStartingDate())
+                .endingDate(request.getEndingDate())
+                .description(request.getDescription())
+                .build();
+
+        return eventRepository.save(eventEntity);
+    }
+
+    public UpdateEventResponse updateEvent(Long eventId, UpdateEventRequest updateEventRequest) {
```

Comment:
Auć :(
1. Można lambdą jak wyżej
2. Wołanie isPresent i get na Optionalach powinno być zabronione przez konwencję genewską
3. nazwa optional jest mocno nie trafiona
4. optional nigdy nie będzie null jeśli to request body tutaj wchodzi.